devito/passes/iet/misc.py

```diff
@@ -116,3 +117,34 @@ def relax_incr_dimensions(iet, **kwargs):
     iet = Transformer(mapper).visit(iet)
 
     return iet, {'efuncs': efuncs}
+
+
+def is_on_device(maybe_symbol, gpu_fit, only_writes=False):
+    """
+    True if all given Functions are allocated in the device memory, False otherwise.
+
+    Parameters
+    ----------
+    maybe_symbol : Indexed or Function or Node
```

Comment:
This var name seems a bit off to me? Something better?

Reply:
I think it's fine, but I'm open to suggestions